math/src/stableswap/math.rs

```diff
@@ -346,6 +347,40 @@ pub(crate) fn calculate_y<const N: u8>(xp: &[Balance], d: Balance, amplification
 	Balance::try_from(y).ok()
 }
 
+pub fn calculate_amplification(
+	initial_amplification: u128,
+	future_amplification: u128,
+	initial_timestamp: u128,
+	future_timestamp: u128,
+	current_timestamp: u128,
+) -> u128 {
+	// short circuit if timestamp are invalid
+	if current_timestamp < initial_timestamp || future_timestamp < initial_timestamp {
+		return initial_amplification;
+	}
+
+	// Short circuit if timestamp or amplification are equal
+	if initial_timestamp == future_timestamp || initial_amplification == future_amplification {
```

Comment:
this block is redundant, the following condition `current_timestamp >= future_timestamp` is satisfied in the same case